source/content/modules-known-issues.md

```diff
@@ -291,6 +291,20 @@ $conf[‘schema_suppress_type_warnings’] = TRUE;
 
 ___
 
+## [Search Api Solr Date Sort](https://www.drupal.org/project/search_api_solr_date_sort)
+
+<ReviewDate date="2020-02-14" />
+
+**Issue**: The module overrides a class from the [Pantheon Apache Solr module](https://pantheon.io/docs/solr-drupal-7) that is responsible for a correct connection to the Pantheon's Apache Solr service. As a result a Solr connection is lost.
+
+**Solution**: Instead of patching the module, you can fix the issue with a custom module:
+
+1. Define a new class that inherits from the `PantheonApachesolrSearchApiSolrService` and contains logic from the `SearchApiSolrDateSortSolrService` (or vice a versa).
+
+1. Implement a `hook_search_api_service_info_alter()` (with a highest weight) and add your class into a configuration array.
```

Comment:
As with the above comment, I don't know if this is all knowledge we can assume  a Drupal user to know, but....

When you say "Implement a `hook_search...`, you don't really have a noun to match with implement. Should that be "function", or the like? Or is that long string something you can refer to as a noun to be implemented?

It sounds like "a highest weight" is bad grammar and I'd have already fixed it, if I was sure that there isn't some Drupal in-the-know shorthand about this.

"into a configuration array"... again, is this something we can assume? 

More generally, is the fix something that would be the same for everyone? If so can we just provide it?

Reply:
The term "implement a hook" is well-known by Drupal developers. If a user is not a Drupal developer - I suspect he will not be able to solve this module's issue by himself.

A module's weight in Drupal is also well-known. But, yeah, we can provide a reference to the article about that on the drupal.org.

"into a configuration array" - I meant here an array that is provided by function (https://git.drupalcode.org/project/search_api_solr_date_sort/-/blob/7.x-1.x/search_api_solr_date_sort.module#L99). Maybe it make sense to clarify that, agree.

About the solution that can be provided for everyone. It can be a standalone module. But I don't think it make sense to create it since because the "search_api_solr_date_sort" module (https://www.drupal.org/project/search_api_solr_date_sort) is not very popular.

Another possible option is to provide a patch for the "search_api_solr_date_sort" module. It would contain only 1 row: replace the "SearchApiSolrService" by the "PantheonApachesolrSearchApiSolrService" in the specifed row https://git.drupalcode.org/project/search_api_solr_date_sort/-/blob/7.x-1.x/includes/service.inc#L10. Maybe it's the easiest way for end users. As for me, I prefer to follow best practices and implement a custom code instead of patching (in a places where it's possible).

So, @alexfornuto what you can advice here? Provide a simple patch OR guide a user how to solve that with a custom code development?

Reply:
> The term "implement a hook" is well-known by Drupal developers. If a user is not a Drupal developer - I suspect he will not be able to solve this module's issue by himself.

OK, so then to make this sentence grammatically correct we would say:

> Implement a `hook_search_api_service_info_alter()` hook...

Or:

> Implement a hook named `hook_search_api_service_info_alter()` ...

Regarding weight, is it "a highest weight", or "the highest weight", or other?

Your logic on not providing the module or patching makes sense to me, thanks for explaining!